src/SSHLibrary/abstractclient.py

```diff
@@ -412,9 +414,8 @@ def read_until_regexp(self, regexp):
         :returns: The read output up and until the `regexp` matches.
         """
         regexp = self._encode(regexp)
-        if isinstance(regexp, basestring):
-            regexp = re.compile(regexp)
-        return self._read_until(lambda s: regexp.search(s), regexp.pattern)
+        regexp = re.compile(regexp)
+        return self._read_until(lambda s: regexp.search(self._encode(s)), regexp.pattern)
```

Comment:
Yet another place where encoding the expected value (`regexp` in this case) should be removed instead of encoding the actual output.